### PR TITLE
CRUD Test suite + updateAll improvements

### LINKED ISSUE #1616: Add CRUDTest & CRUDAutoIncTest to test functionality on each database we support
From this comment:

https://github.com/bitcoin-s/bitcoin-s/pull/1611#issuecomment-650128425

We really need tests that directly test all functionality in `CRUD`/`CRUDAutoInc` _without_ testing other functionality related to a specific DAO. Hopefully these will save us a lot of time in the future of trying to figure out "why things are working on sqlite, but not on Posgres". We should be able to do this for all database drivers we support. As of the time of this writing, we support postgre and sqlite

### PR DESCRIPTION
**Pull request by benthecarman**
Closes #1616 

Adds tests for the `CRUD` interface.

### CODE REVIEW

File: db-commons/src/main/resources/db.conf [hunk 1]
```diff
@@ -51,4 +51,13 @@ bitcoin-s {
       url = "jdbc:sqlite:"${bitcoin-s.chain.db.path}${bitcoin-s.chain.db.name}
     }
   }
+
+  test = ${common}
```
Comment (Christewart) [inline db-commons/src/main/resources/db.conf]: This probably should not be in `src/main`, the configuration should be in `test`

File: testkit/src/main/resources/postgresql/test/migration/V1__test_db_baseline.sql [hunk 2]
```diff
@@ -0,0 +1 @@
+CREATE TABLE test_table (pk TEXT NOT NULL,data TEXT NOT NULL,constraint pk_pk primary key(pk));
```
Comment (Christewart) [inline testkit/src/main/resources/postgresql/test/migration/V1__test_db_baseline.sql]: I don't think we need migrations for this do we? This is strictly for test cases so I think we can just call create the table directly with Slick using `DbManagement.createTable()`?

**Review by Christewart — COMMENTED** [review]
So does this iteration strictly only test postgres or does it test sqlite as well? If it tests both, where exactly is the code that differentiates which database driver we are testing?

**Comment by benthecarman** [comment]
> So does this iteration strictly only test postgres or does it test sqlite as well? If it tests both, where exactly is the code that differentiates which database driver we are testing?

This is similar to how our other tests work and needs `PG_ENABLED="1"` to be postgres. I would like to make it test both without having to do that but haven't fully figured that out yet.

File: testkit/src/main/resources/postgresql/test/migration/V1__test_db_baseline.sql (hunk 2 above)
Comment (benthecarman) [reply testkit/src/main/resources/postgresql/test/migration/V1__test_db_baseline.sql]: I think this is preferred, this way we can test that the migrations are correctly hooked up to the tests

Comment (Christewart) [reply testkit/src/main/resources/postgresql/test/migration/V1__test_db_baseline.sql]: This is just test code isn't it? We don't need migrations for test code IMO as we can just re-create the test environment

File: testkit/src/main/scala/org/bitcoins/testkit/db/TestDAOFixture.scala [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+                                       TestDb("abc3", hex"0053"))
+
+  val updatedDb: TestDb = testDb.copy(data = hex"0000")
+
+  val updatedDbs: Vector[TestDb] = testDbs.map(_.copy(data = hex"0000"))
+
+  def testCreate(testDAO: TestDAO): Future[Boolean] = {
+    for {
+      _ <- testDAO.create(testDb)
+      read <- testDAO.read(id = "abc")
```
Comment (Christewart) [inline testkit/src/main/scala/org/bitcoins/testkit/db/TestDAOFixture.scala]: nit: `testDb.pk` no?

**Review by Christewart — APPROVED** [review]

**Pull request merged**